### PR TITLE
More detailed build instructions, closes #107

### LINKED ISSUE #107: Dockerfile build fails
Building the library using `builda bud` as explained in the docs 
https://github.com/containers/fuse-overlayfs/blob/f8ba9ade20d107d3e6a3dc11b6e01116894d2387/README.md#L36

fails at [this line](https://github.com/containers/fuse-overlayfs/blob/4dc60f0e157293f8692446e35d3bfc7420e36773/Dockerfile.static#L12) with error:
```
[...]
Installing /build/libfuse/doc/mount.fuse3.8 to /usr/share/man/man8
Installing /build/libfuse/build/meson-private/fuse3.pc to /usr/lib64/pkgconfig
Running custom install script '/build/libfuse/util/install_helper.sh /etc /usr/bin /usr/lib/udev/rules.d true'
mknod: /dev/fuse: Operation not permitted
FAILED: meson-install
/usr/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
```

As far as I understand the container runtime (rightfully) prevents [this call](https://github.com/libfuse/libfuse/blob/f0e08cc700d629da2d46def8b620b0ed858cc0d9/util/install_helper.sh#L35) to `mknod`. If run in privileged mode, the call would probably succeed, but that is probably unsafe and will likely modify the host; on the other hand, this might be desired (in my case, the host already has FUSE2 installed, so it makes no difference).

Instead, I simply "tricked" the script into [believing](https://github.com/libfuse/libfuse/blob/f0e08cc700d629da2d46def8b620b0ed858cc0d9/util/install_helper.sh#L33) that the device already exists by adding the line `touch /dev/fuse` before [this one](https://github.com/containers/fuse-overlayfs/blob/4dc60f0e157293f8692446e35d3bfc7420e36773/Dockerfile.static#L12). With this workaround the build completes successfully.

Further problems I encountered:
* [this command](https://github.com/containers/fuse-overlayfs/blob/4dc60f0e157293f8692446e35d3bfc7420e36773/Dockerfile.static#L21) does not seem to work: `fuse-overlayfs: upperdir not specified`; as far as I understand, this is expected.
* it might be helpful to newcomers to explain what to do after the build (grab the binary from the image and install it somewhere in the PATH on the host).

### PR DESCRIPTION
**Pull request by matpen**
Just a few improvements to the `README.md` in order to facilitate newcomers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,16 +22,49 @@ Specify a different UID/GID mapping:
 $ fuse-overlayfs -o uidmapping=0:10:100:100:10000:2000,gidmapping=0:10:100:100:10000:2000,lowerdir=lowerdir/a:lowerdir/b,upperdir=up,workdir=workdir merged
 ```
 
-Build Requirements:
+Requirements:
 =======================================================
 
-fuse-overlayfs requires libfuse > v3.2.1
+Please note that `fuse-overlayfs` requires Linux Kernel > v4.8.0.
```
Comment (giuseppe) [inline README.md]: why is Linux v4.8.0 required?

Comment (caemate) [reply README.md]: I faced this problem myself at the time of first install, and finally spotted [your comment](https://github.com/containers/libpod/issues/3109#issuecomment-491848145) that clarifies this.

In addition, there is a typo in my code, which I now fixed!

Comment (matpen) [reply README.md]: I faced this problem myself at the time of first install, and finally spotted [your comment](https://github.com/containers/libpod/issues/3109#issuecomment-491848145) that clarifies this.

In addition, there is a typo in my code, which I now fixed!

**Comment by giuseppe** [comment]
I see thanks!  Could you drop that part completely or at least specify (to work in a user namespace)?

Older versions of Linux are still fine if you don't use fuse-overlayfs from a user namespace (such as rootless Podman).

Nit, could you squash the patches together?

**Comment by matpen** [comment]
Done! I tried to rephrase to include that precious information. Let me know your thoughts, and I can refine if necessary.

> Nit, could you squash the patches together?

Sure, it should look like one clean change now.

**Comment by giuseppe** [comment]
Thanks, LGTM!

**Pull request merged**